src/labone/mock/automatic_session_functionality.py

```diff
@@ -0,0 +1,267 @@
+
+        Warning:
+            Flags will be ignored in this implementation. (TODO)
+            For now, the behaviour is equivalent to
+            ListNodesFlags.RECURSIVE | ListNodesFlags.ABSOLUTE
+
+        Args:
+            path: Path to narrow down which nodes should be listed. Omitting
+                the path will list all nodes by default.
+            flags: Flags to control the behaviour of the list_nodes method.
+
+        Returns:
+            List of paths.
+        """
+        if path in [""]:
+            return []
+        return [
+            p
+            for p in self.memory
+            if fnmatch.fnmatch(p, path)
+            or fnmatch.fnmatch(p, path + "*")
+            or fnmatch.fnmatch(p, path + "/*")
+            or p == path
+        ]
+
+    async def get(self, path: LabOneNodePath) -> AnnotatedValue:
+        """Predefined behaviour for get.
+
+        Look up the path in the internal dictionary.
+
+        Args:
+            path: Path of the node to get.
+
+        Returns:
+            Corresponding value.
+        """
+        try:
+            value = self.memory[path].value
+        except KeyError as e:
+            msg = f"Path {path} not found in mock server. Cannot get it."
+            raise LabOneMockError(msg) from e
+        response = AnnotatedValue(path=path, value=value)
+        response.timestamp = self.get_timestamp()
+        return response
+
+    async def get_with_expression(
+        self,
+        path_expression: LabOneNodePath,
+        flags: ListNodesFlags  # noqa: ARG002
+        | int = ListNodesFlags.ABSOLUTE
+        | ListNodesFlags.RECURSIVE
+        | ListNodesFlags.LEAVES_ONLY
+        | ListNodesFlags.EXCLUDE_STREAMING
+        | ListNodesFlags.GET_ONLY,
+    ) -> list[AnnotatedValue]:
+        """Predefined behaviour for get_with_expression.
+
+        Find all nodes associated with the path expression
+        and call get for each of them.
+
+        Args:
+            path_expression: Path expression to get.
+            flags: Flags to control the behaviour of the get_with_expression method.
+
+        Returns:
+            List of values, corresponding to nodes of the path expression.
+        """
+        return [await self.get(p) for p in await self.list_nodes(path=path_expression)]
+
+    async def set(self, value: AnnotatedValue) -> AnnotatedValue:  # noqa: A003
+        """Predefined behaviour for set.
+
+        Updates the internal dictionary. A set command is considered
+        as an update and will be distributed to all registered subscription handlers.
+
+        Args:
+            value: Value to set.
+
+        Returns:
+            Acknowledged value.
+        """
+        if value.path not in self.memory:
+            msg = f"Path {value.path} not found in mock server. Cannot set it."
+            raise LabOneMockError(msg)
+        self.memory[value.path].value = value.value
+
+        timestamp = self.get_timestamp()
+
+        capnp_response = {
```

Comment:
suggestion: Since this is a rather expensive operation, wouldn`t it make sense to check if there is even a subscription before converting the value?